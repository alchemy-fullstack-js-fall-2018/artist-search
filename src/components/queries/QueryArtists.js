import React from 'react';
import PropTypes from 'prop-types';

const QueryArtists = ({ query, onQueryItemChange }) => {

  return (
    <form>
      <label htmlFor="name">Artist name: </label>
      <input type="text" name="name" value={query.name} onChange={onQueryItemChange}/>
    </form>
  );


};

QueryArtists.propTypes = {
  query: PropTypes.object.isRequired,
  onQueryItemChange: PropTypes.func.isRequired
};

export default QueryArtists;
