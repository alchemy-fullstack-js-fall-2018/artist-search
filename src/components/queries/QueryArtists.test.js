import React from 'react';
import { shallow } from 'enzyme';
import QueryArtists from './QueryArtists';

describe ('tests for QueryArtists component', () => {

  it('renders an QueryArtists component', () => {
    const onQueryItemChangeFn = jest.fn();
    const query = {
      name: 'Joni Mitchell'
    };
    const wrapper = shallow(
      <QueryArtists query={query}
        onQueryItemChange={onQueryItemChangeFn}
      />);
    expect(wrapper).toMatchSnapshot();
  });

});
