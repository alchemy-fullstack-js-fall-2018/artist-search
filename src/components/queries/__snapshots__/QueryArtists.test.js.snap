// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for QueryArtists component renders an QueryArtists component 1`] = `
<form>
  <label
    htmlFor="name"
  >
    Artist name: 
  </label>
  <input
    name="name"
    onChange={[MockFunction]}
    type="text"
    value="Joni Mitchell"
  />
</form>
`;
